src/Repository/ContributorRepository.php

```diff
@@ -40,6 +40,24 @@ public function remove(Contributor $entity, bool $flush = false): void
         }
     }
 
+    public function checkIfExistAndSave(array $contributors, Project $project): void
+    {
+        foreach ($contributors as $contributorData) {
+            $contributor = $this->findOneBy(['githubName' => $contributorData['login']]);
+            if (!$contributor) {
+                $contributor = new Contributor();
+                $contributor->setName($contributorData['login']);
+                $contributor->setGithubAccount($contributorData['html_url']);
+                $contributor->setGithubName($contributorData['login']);
+                $this->save($contributor, true);
+            }
+            if (!$contributor->getProjects()->contains($project)) {
+                $contributor->addProject($project);
+                $this->save($contributor, true);
```

Comment:
Why have you two $this->save() ? you only should have one to not make two request